docs/reference/remap/match.cue

```diff
@@ -0,0 +1,47 @@
+package metadata
+
+remap: functions: match: {
+	arguments: [
+		{
+			name: "value"
+			description: "The value to match."
+			required: true
+			type: ["string"]
+		},
+		{
+			name: "pattern"
+			description: "The regular expression pattern to match against."
+			required: true
+			type: ["regex"]
+		}
+	]
+	return: ["bool"]
+	category: "text"
+	description: """
+		Determines whether a string matches the provided pattern.
+		"""
+	examples: [
+		{
+			title: "Successful match"
+			input: {
+				value: "I'm a little teapot"
+				pattern: "/teapot/"
+			}
+			source: ".matches = match(.value, .pattern)"
```

Comment:
This won't compile. The pattern can't be specified in the input as it is costly to compile the regex.

We force the regex to be hardcoded in the script like:

```coffescript
.matches = match(.value, /teapot/)
```

Reply:
Ah, okay, I thought maybe the web template would swap those out. Good to know, will fix.